### PR TITLE
refactor(swift): Enable strict null checks

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Fix strict null errors

## Context:

- Ref: #7154
- Waiting for: #12842

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by zharinov** [comment]
Waiting for #12866

**Comment by zharinov** [comment]
Ref: #12869

File: tsconfig.strict.json [hunk 1]
```diff
@@ -131,6 +131,8 @@
     "./lib/util/regex.ts",
     "./lib/util/sanitize.ts",
     "./lib/util/split.ts",
+    "./lib/versioning/swift/index.spec.ts",
+    "./lib/versioning/swift/index.ts",
```
Comment (rarkins) [inline tsconfig.strict.json:135]: How about range?

Comment (zharinov) [reply tsconfig.strict.json:135]: Good idea, I'll convert `files` to `include` once current PRs are done

File: lib/versioning/swift/index.spec.ts [hunk 2]
```diff
@@ -86,6 +90,7 @@ describe('versioning/swift/index', () => {
   `(
     'isLessThanRange("$version", "$range") === "$expected"',
     ({ version, range, expected }) => {
+      assert(isLessThanRange);
```
Comment (viceice) [inline lib/versioning/swift/index.spec.ts:93]: Why assert?

File: lib/versioning/swift/index.ts [hunk 3]
```diff
@@ -29,21 +29,36 @@ const {
...
-const matches = (version: string, range: string): boolean =>
-  satisfies(version, toSemverRange(range));
+
+function getSatisfyingVersion(
+  versions: string[],
+  range: string
+): string | null {
+  const normalizedVersions = versions.map((v) => v.replace(regEx(/^v/), ''));
+  const semverRange = toSemverRange(range);
+  return semverRange && maxSatisfying(normalizedVersions, semverRange);
```
Comment (viceice) [inline lib/versioning/swift/index.ts:41]: I think this now returns `false` instead of `null` ?

File: lib/versioning/swift/index.spec.ts (hunk 2 above)
Comment (zharinov) [reply lib/versioning/swift/index.spec.ts:93]: At the morning, it seemed to me the most elegant way to tell compiler the optional function is really exists for this particular `VersioningApi`  instance 😅

File: lib/versioning/swift/index.ts (hunk 3 above)
Comment (zharinov) [reply lib/versioning/swift/index.ts]: `semverRange` is either string or null, though made it clear with code anyway

**Review by viceice — APPROVED** [review]

**Pull request merged**